src/bin/client.rs

```diff
@@ -0,0 +1,110 @@
+use anyhow::{anyhow, bail};
```

Comment:
The server binary has a `run` command that runs the binary with a local probe, it's useful to run a binary exactly like the teleprobe server would (for example probe-run doesn't support booting from RAM). I use that quite often, it would make sense to have it on the "client" binary as well (or just unify the binaries, or move it to the client binary)

I wonder if it'd be better to make everything a single binary? The only advantage I can think of for separate binaries is a smaller binary if you only want the client, but if it's already pulling the whole tokio (and probe-rs, for local run) then it won't be much smaller.

Perhaps with all "local" stuff could be under a subcommand:

```
teleprobe local run --elf blah --probe blah
teleprobe local list-probes
teleprobe server
teleprobe client list-targets
teleprobe client run --elf blah --target blah